### ISSUE TITLE
Unexpected behavior when setting multiple navigations

### ISSUE THREAD

**Issue by anis-gist** [dotnet/efcore#28512]
## File a bug

Getting some unexpected behavior when setting multiple navigations to the same table in the same DB context.
Assuming I have 3 tables: party, address, and country. Both party and address have a foreign key to the country table. I am setting the two foreign keys in the same context as seen below.

### Include your code
```C#
using (var context = new someContext())
                {
                    Person person = context.person.Where(c => c.id == person_id).FirstOrDefault();
                    Address address = context.address.Where(c => c.id == address_id).FirstOrDefault();
                    Console.WriteLine(person.nationality_code);
                    Console.WriteLine(address.country_code);
                    address.country_codeNavigation = context.country.Where(t => t.code == "USA").FirstOrDefault();
                    person.nationality_codeNavigation = context.country.Where(t => t.code == "UK").FirstOrDefault();
                    context.SaveChanges();
                    Console.WriteLine(person.nationality_code);
                    Console.WriteLine(address.country_code);
                }
```
### Unexpected Output
The above code doesn't work as expected if the country code (of address) and nationality code (of person) are **the same** when they get fetched. e.g. Assuming both country_code and nationality_code were "UK" upon fetch, the following output is yielded
```
UK
UK
UK
UK
```
The expected output is 
```
UK
UK
UK
USA
```
### Include provider and version information

EF Core version: 6.0.7
Database provider: Npgsql.EntityFrameworkCore.PostgreSQL 6.0.5
Target framework: .NET 6.0
Operating system: Windows 10
IDE: Visual Studio 2022 17.2.6

**Comment by ajcvickers** [comment]
This issue is lacking enough information for us to be able to fully understand what is happening. Please attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing so that we can investigate.

**Comment by anis-gist** [comment]
Attached is a simple console app replicating my env.

It assumes a Postgres db called "test" is present. Not sure if there is a better way for me to create such a demo project.

[ConsoleApp3.zip](https://github.com/dotnet/efcore/files/9189807/ConsoleApp3.zip)

**Comment by anis-gist** [comment]
Corrected project

[ConsoleApp3.zip](https://github.com/dotnet/efcore/files/9201053/ConsoleApp3.zip)

**Comment by ajcvickers** [comment]
Duplicate of #27497. This is fixed in the [daily builds](https://github.com/dotnet/efcore/blob/main/docs/DailyBuilds.md) and will be included in EF Core 7.0 preview 7.

Without the enhancement in #27497, a call to `DetectChanges` is need after changing the navigation property:

```C#
    address1.country_codeNavigation = address1CountryCodeNavigation;

    context.ChangeTracker.DetectChanges();
```

**Comment by anis-gist** [comment]
Thank you will check these resources out.
